api/inc/vmpu_exports.h

```diff
@@ -153,59 +153,65 @@ typedef struct {
     UvisorBoxAcl acl;
 } UVISOR_PACKED UvisorBoxAclItem;
 
+/* This struct contains all the BSS sections that uVisor allocates for a secure
+ * box. It can be used to keep the sizes of the sections or their pointers. */
+typedef struct uvisor_bss_sections_t {
```

Comment:
This structure is used both to hold sizes:

```C
union {
    size_t sizes[UVISOR_BSS_SECTIONS_COUNT];
    UvisorBssSections size_of;
} const bss;
```

and pointers:

union {
    void * pointers[UVISOR_BSS_SECTIONS_COUNT];
    UvisorBssSections address_of;
} const bss;
```

so I used `uint32_t` as base type.

If you are referring to the RPC casting that would be needed anyway, even with a `void *`, but I think there are better ways to fix the RPC specific problem. The reason I didn't fix it now is because we will anyway need to change that code. What other casting were you thinking about?